### PR TITLE
BLD: depend on lightpath-base to avoid ui dependencies in builds

### PR DESCRIPTION
**Pull request by tangkong**
## Description
- Swaps `lightpath` -> `lightpath-base` in conda recipe
- Specifies >1.0.5 for both, to ensure pypi builds pull from version with GUI extras separated
  - Conda builds will fail until the feedstock updates, but I figured it's better to keep the version numbers the same

## Motivation and Context
Stop depending on things we don't need!

## How Has This Been Tested?
go CI go

## Where Has This Been Documented?
This PR

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
fwiw lightpath-base is currently only up to 1.0.4 on conda-forge so maybe this will need to wait until the tag flows through the CI chain

**Comment by tangkong** [comment]
I believe the dependency to look for here is typhos.  (As in typhos should not be installed, as it is a gui-only dependency of lightpath)

**Comment by tangkong** [comment]
A local pip-install successfully pulls in lightpath without gui dependencies, and the conda build here pulls in lightpath-base

I was sort of hoping to get pcdsdevices to build without qt as a dependency, but as long as typhos is a test-dependency we won't see it in CI.  I think it's probably easiest to test this locally after this is merged.  

I still think the changes here are correct, regardless of whether or not qt is fully extricated.

**Comment by ZLLentz** [comment]
Confirmed that the changes look good
Confirmed that typhos is actually used in the test suite and can't be removed without stripping out some tests

Does this need a pre-release notes stub?

**Review by ZLLentz — APPROVED** [review]

**Comment by ZLLentz** [comment]
fwiw the main thing typhos does in the test suite is verify that it can see our entrypoints, if you want to think a bit about how valuable or forgettable that might be.

**Comment by tangkong** [comment]
I think the test suite requiring typhos is fine, I just wanted to make sure that normal installs don't pull in more dependencies than needed.  The CI just isn't the place to do that I think, and that's ok

**Pull request merged**
